### PR TITLE
Fix: Update advanced dhcp server to new nic syntax

### PR DESCRIPTION
**Pull request by oxedions**
Update advanced dhcp server to use new NIC syntax in inventory.

### CODE REVIEW

File: roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2 [hunk 1]
```diff
@@ -39,40 +39,40 @@
...
   host {{host}}-{{item}} { 
     option host-name "{{host}}";
-    hardware ethernet {{nic_args.mac}};
-    fixed-address {{nic_args.ip4}};
+    hardware ethernet {{nic.mac}};
+    fixed-address {{nic.ip4}};
     filename "{{ipxe_rom_filename}}";
   }
-        {% elif nic_args.dhcp_client_identifier is defined and not none %}
+        {% elif nic.dhcp_client_identifier is defined and not none %}
```
Comment (actatux) [inline roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2:52]:
```suggestion
        {% elif nic.dhcp_client_identifier is defined %}
```

File: roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2 [hunk 2]
```diff
@@ -39,40 +39,40 @@
...
   host {{host}}-{{item}} {
     option host-name "{{host}}";
-    option dhcp_client_identifier {{nic_args.dhcp-client-identifier}};
-    fixed-address {{nic_args.ip4}};
+    option dhcp_client_identifier {{nic.dhcp-client-identifier}};
```
Comment (actatux) [inline roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2:55]: Avoid dashes in variable names.
```suggestion
    option dhcp_client_identifier {{nic.dhcp_client_identifier}};
```

File: roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2 [hunk 3]
```diff
@@ -39,40 +39,40 @@
...
+    fixed-address {{nic.ip4}};
     filename "{{ipxe_rom_filename}}";
   }
-        {% elif nic_args.host_identifier is defined and not none %}
+        {% elif nic.host_identifier is defined and not none %}
```
Comment (actatux) [inline roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2:59]:
```suggestion
        {% elif nic.host_identifier is defined %}
```

File: roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2 [hunk 4]
```diff
@@ -39,40 +39,40 @@
...
   host {{host}}-{{item}} {
     option host-name "{{host}}";
-    host_identifier {{nic_args.host-identifier}};
-    fixed-address {{nic_args.ip4}};
+    host_identifier {{nic.host-identifier}};
```
Comment (actatux) [inline roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2:62]:
```suggestion
    host_identifier {{nic.host_identifier}};
```

File: roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2 [hunk 5]
```diff
@@ -39,40 +39,40 @@
...
+    fixed-address {{nic.ip4}};
     filename "{{ipxe_rom_filename}}";
   }
-        {% elif nic_args.match is defined and not none %}{# Warning, only works in a shared network!! #}
+        {% elif nic.match is defined and not none %}{# Warning, only works in a shared network!! #}
```
Comment (actatux) [inline roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2:66]:
```suggestion
        {% elif nic.match is defined %}{# Warning, only works in a shared network!! #}
```

**Comment by oxedions** [comment]
Dear @actatux 

Thank you for this review. I will use your propositions as commit 😊

**Pull request merged**